### ISSUE TITLE
A call whose name opens on a code point outside ASCII is read as an address by the scan that finds comments

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#343]
### What minimal example or steps are needed to reproduce the bug?

`findCommentSpans` decides whether a `url(` it has found opens a token of its own or closes the name of some other call, and it asks that with `WORD_BRACE_OR_HYPHEN`, whose `\w` is ASCII and nothing else. A name whose character in front of the address is outside ASCII is therefore no name to it, and the call is read as an address — its double slashes carried as ordinary characters, the rest of the line left as code.

Under `customSyntax: "postcss-less"`, with `--fix`:

| rule | option | code | what comes back |
|---|---|---|---|
| `unit-case` | `"lower"` | `a { b: éurl(http://a/b.png) 1PX; }` | `a { b: éurl(http://a/b.png) 1px; }` |
| `unit-case` | `"lower"` | `a { b: aurl(http://a/b.png) 1PX; }` | nothing reported, nothing written |

The two differ in one character of a name and in nothing else. `lightningcss` reads both as ordinary calls, leaving `éurl(http://a/b.png)` and `aurl(http://a/b.png)` exactly as they stand rather than normalising either to a `url()`, so under Less the `//a/b.png) 1PX; }` of each is the text of a comment and the `1PX` in it is nothing a rule may write to. The second row is right and the first is the same false positive [#271](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/271) is the census of.

The scan says as much on its own:

```js
findCommentSpans(`éurl(http://x/y.png)`)   // []           — read as an address
findCommentSpans(`aurl(http://x/y.png)`)   // [{ … }]      — read as a call
```

`日本url(http://x/y.png)` and `·url(http://x/y.png)` go the way of the first.

### What minimal configuration is needed to reproduce the bug?

```json
{
  "customSyntax": "postcss-less",
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/unit-case": "lower"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.less" --config myconfig.json --fix
```

### What did you expect to happen?

A call named with any identifier CSS allows is a call, and its arguments may hold a comment. An identifier may hold code points outside ASCII, which is one of the three directions [#252](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/252) already widened `findFunctionArgumentSpans` in.

### What actually happened?

Only a name spelled in ASCII word characters, a closing brace or a hyphen is seen, so a call named otherwise is read as an address and everything its arguments hold is read as code.

### What is the cause?

`WORD_BRACE_OR_HYPHEN` is `/[\w}-]/u`, and `\w` in a JavaScript expression is `[A-Za-z0-9_]` whatever flags stand behind it. The sibling scan asks the same question with `IDENTIFIER_CODE_POINT`, which is the wide reading; this one has never had it, on either side of [#321](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/321), which moved the question from a look back at one character to an answer carried forwards without touching the predicate itself.

### Do you have a proposal to fix the bug?

Ask the wide reading here too. The flag the scan carries is set in one place, so the change is to that predicate and to nothing else — but it changes what every reader of `findCommentSpans` reads, so it wants its own branch and its own sweep, the way #321 did.
